include/class.ticket.php

```diff
@@ -1212,7 +1212,7 @@ function getMissingRequiredField() {
     function addCollaborator($user, $vars, &$errors, $event=true) {
 
         if ($user && $user->getId() == $this->getOwnerId())
-            $errors['err'] = __('Ticket Owner cannot be a Collaborator');
+            return null;
```

Comment:
I think this is needed when adding collaborators via UI. You shouldn't be attempting to add  ticket owner as collaborator anyhow - perhaps that's where you need to do the check to avoid the error.